client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-large-preview.tsx

```diff
@@ -135,10 +137,61 @@ const PatternLargePreview = ( {
 		);
 	};
 
+	const deviceComputerViewportScale = () =>
+		frameRef.current!.clientWidth / DEVICE_COMPUTER_VIEWPORT_WIDTH;
+
+	const setLargePreviewStyleForViewport = () => {
+		if ( NAVIGATOR_PATHS.SECTION_PATTERNS === navigator.location.path ) {
+			// Extend state with viewport styles
+			setPatternLargePreviewStyle(
+				( state ) =>
+					( {
+						...state,
+						'--pattern-large-preview-device-computer-width': DEVICE_COMPUTER_VIEWPORT_WIDTH,
+						'--pattern-large-preview-device-computer-scale': deviceComputerViewportScale(),
+					} as CSSProperties )
+			);
+		}
+	};
+
 	const updateViewportHeight = () => {
-		setViewportHeight( frameRef.current?.clientHeight );
+		let height = frameRef.current?.clientHeight as number;
+		if ( 'computer' === device ) {
+			// Scale up for patterns with 100vh
+			height = height / deviceComputerViewportScale();
+		}
+		setViewportHeight( height );
 	};
 
+	// Scale down the computer device viewport
+	useEffect( () => {
+		if ( NAVIGATOR_PATHS.SECTION_PATTERNS === navigator.location.path ) {
+			setTimeout(
+				() => {
+					updateViewportHeight();
+					setLargePreviewStyleForViewport();
+				},
+				// Wait for .pattern-large-preview transition
+				205
+			);
+		}
+	}, [ navigator.location ] );
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg)

If we just want to execute a function in a certain condition, wouldn't it be ok to remove useEffect? (I'm sorry if it doesn't make sense here.)
https://react.dev/learn/you-might-not-need-an-effect

```suggestion
		if ( NAVIGATOR_PATHS.SECTION_PATTERNS === navigator.location.path ) {
			setTimeout(
				() => {
					updateViewportHeight();
					setLargePreviewStyleForViewport();
				},
				// Wait for .pattern-large-preview transition
				205
			);
		}
```

Reply:
I noticed that the value `navigator.location` was not updated when is out of the effect. I'll check it again.